### ISSUE TITLE
Transparency issues

### ISSUE THREAD

**Issue by joereynolds** [joereynolds/Mr-Figs#46]
Ever since moving over to pyscroll for the scrolling camera, we've lost transparency. Probably user error.

**Comment by bitcraft** [comment]
i took a look at your transparency issue and found it.  it was an odd one...  if you look at the tmx level files, you'll notice that the tileset images have a attribute called "trans" set.
```xml
  <image source="../../newtiledsheet.png" trans="ff00ff" width="144" height="224"/>
```
this is supposed to mean "colorkey".  what is happening here is that the maps are loaded with pytmx as 32-bit images with an alpha channel.  pyscroll will reduce all the tiles when loaded so it draws faster, if possible.  because tiled has a colorkey set (the trans attribute), pyscroll tries to treat the tiles as having a color key, so the alpha channel is lost and leaves some tiles with an ugly border.

the `alpha` kwarg for the bufferered renderer isn't obvious.  all tiles will have an alpha channel if they require it, but the layers do not.  this flag means "give the layers an alpha channel".  enabling reduces performance by about 30% because there are 8 more bits to write per pixel for each layer, so it is disabled by default.  it is only needed for parallax scrolling and other effects.

TL;DR - your tileset is misconfigured to use a color key when it really has a alpha channel
* edit the tmx files and remove the color key attribute ("trans")
* remove the "alpha=True" kwarg from the pyscroll buffered renderer

**Comment by joereynolds** [comment]
Wow @bitcraft , I can't thank you enough for the support you've given me with tmx/tiled related things and then I wake up to you doing a full-on investigation that would've taken me much longer.

I've done as you suggested and it's working wonderfully now, I'll have to look at the configuration of my tileset and see what's going on. 

Is there a way to donate to pytmx/pyscroll? It's saved me a lot of time (as have you)
Thanks again!

**Comment by bitcraft** [comment]
thank you for your generosity.  i can accept paypal at leif.theden@gmail.com and i also have librepay https://liberapay.com/ltheden/.  glad this has helped you!  that honetly is reward enough, anything more than that is very appreciated.
